投稿機能の追加

arima0714 opened the pull request:
## チケットへのリンク
* なし

## やったこと
* ユーザー IDが1として、投稿されるようになった
* 投稿時に自動で日付が入るようになった
* 初期化用のMakeコマンドを追加した

## やらないこと
* 閲覧機能

## できるようになること（ユーザ目線）
* 投稿が可能になる

## できなくなること（ユーザ目線）
* なし

## 動作確認

1. 起動 [やりかた](https://qiita.com/t-iguchi/items/ba323f4d418d158d2116#5-%E5%AE%9F%E8%A1%8C%E3%81%97%E3%81%A6%E7%A2%BA%E8%AA%8D%E3%81%99%E3%82%8B)
2. `make init`
3. [ここ](http://localhost:8080/post) にアクセス
4. テキストボックスに何かを入力する
5. Post をクリック
6. postsテーブルに投稿内容が入る

## その他
### 特にレビューしてもらいたいところ
* なし

### 残タスク(余裕があったら)
* 閲覧機能

yumemi-nshibata commented on src/main/kotlin/com/example/demo/repository/PostRepository.kt:
```diff
@@ -0,0 +1,9 @@
+
+@Repository
+interface PostRepository : JpaRepository<Post, Int> {
+}
```
最終行に改行がある場合と、ない場合があるのでどちらに統一してほしいです。
(Intellijの機能で、保存時に整形してくれる機能があるので、それを利用すると良いかもしれません)

yumemi-nshibata commented on src/main/kotlin/com/example/demo/model/Post.kt:
```diff
@@ -0,0 +1,20 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "post_id", nullable = true, updatable = false)
+    var postId: Int? = 1,
+    @Column(name = "user_id", nullable = false, updatable = false)
```
ちょっと見づらいので、
変数の定義とアノテーションの間は改行がほしいです

yumemi-nshibata commented on src/test/kotlin/com/example/demo/DemoApplicationTests.kt:
```diff
@@ -1,13 +1,13 @@
-package com.example.test-project
+package com.example.SaladBoard
```
パッケージ名とファイル階層を一致させたほうがよいです。

yumemi-nshibata commented on src/main/resources/application.properties:
```diff
@@ -1 +1,8 @@
+spring.datasource.url=jdbc:mysql://localhost:3306/db
+spring.datasource.username=root
+spring.datasource.password=root
+spring.jpa.hibernate.ddl-auto=update
```
値は `none` でお願いします

yumemi-nshibata commented on src/main/kotlin/com/example/demo/request/PostForm.kt:
```diff
@@ -0,0 +1,3 @@
+package com.example.demo.request
```
パッケージ名の修正をお願いします

yumemi-nshibata commented on src/main/kotlin/com/example/demo/service/PostService.kt:
```diff
@@ -0,0 +1,14 @@
+
+@Service
+class PostService(val postRepository: PostRepository) {
+    fun savePost(body: PostForm) {
```
関数の説明がほしいです

yumemi-nshibata commented on src/main/resources/templates/post.html:
```diff
@@ -0,0 +1,21 @@
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
+    <link rel="stylesheet" th:href="@{/webjars/bootstrap/3.3.7-1/css/bootstrap.min.css}"/>
+    <meta name="viewport" content="width=device-width, initial-scale=1"/>
+    <title>Post</title>
```
SaladBoard

yumemi-nshibata commented on Makefile:
```diff
@@ -0,0 +1,14 @@
+.PHONY: insert_users
+insert_users:
+	mysql --host=127.0.0.1 --port=3306 --user=user --password=password -e "use db; INSERT INTO users (email, password) VALUES (1, 1);"
+	mysql --host=127.0.0.1 --port=3306 --user=user --password=password -e "use db; INSERT INTO users (email, password) VALUES (2, 2);"
+	mysql --host=127.0.0.1 --port=3306 --user=user --password=password -e "use db; INSERT INTO users (email, password) VALUES (3, 3);"
+	mysql --host=127.0.0.1 --port=3306 --user=user --password=password -e "use db; INSERT INTO users (email, password) VALUES (4, 4);"
```
flywayに移動してください

arima0714 committed:
開発環境関連の修正

.editorconfigを追加し、ファイル末改行が行われる
PULL_REQUEST_TEMPLATE.mdを追加した
.gitignoreに追跡する必要のないファイルを更新

arima0714 committed:
DBの接続設定と初期化に関する修正

DBの接続設定を修正
DBの初期化時にflywayによるデータ投入が実施されるように修正

arima0714 committed:
プロジェクトの設定を追加

依存関係を追加
DB接続設定を追加

arima0714 committed:
投稿機能の追加

Model->Post.kt
View->post.html
Controller->PostController.kt
Service->PostService.kt
Repository->PostRepository.kt

プロジェクト名を修正(demo->SaladBoard)

yumemi-nshibata commented on src/main/kotlin/com/example/SaladBoard/service/PostService.kt:
```diff
@@ -0,0 +1,15 @@
+
+@Service
+class PostService(val postRepository: PostRepository) {
+    //    受け取った入力を保存する
```
```(suggest)
/**
* 受け取った入力を保存する
*/
```

yumemi-nshibata replied:
kotlin/java特有の書き方なのですが、、
上記のように書くとドキュメントとして扱われます

参考: https://maku77.github.io/kotlin/misc/kdoc.html

arima0714 committed:
関数のコメントの形式を変更

"//"を利用したコメントから、KDoc形式に修正
